### DISCUSSION TITLE
Baklava attestation bot

### DISCUSSION THREAD

**Discussion by aaronmboyd** [celo-org/social-connect#242] [Help (Q&A)]
How does this work? Is it simply reporting success 100% of the time?

I'm working on some new features for Celo Vido visualizers and after indexing attestation events in Baklava, it appears like not only does each attestation request send many more AttestationIssuerSelected events that I would have expected on mainnet, 100% of them appear to be completed.

Back in earlier epochs, it looks like some were failing / incomplete as I would expect, for example epoch 117 has a ton of failures. However recently, everything looks super consistent. Where all registered attestors get nearly the same amount of requests and all succeed.

Has the logic changed somehow? I just want to make sure I'm indexing the correct information, but from the emitted Events it looks like I am...?

For example, epoch 720, shows 13 attestations for each validator currently, and 100% completion rate. 
![image](https://user-images.githubusercontent.com/13218496/199224260-a025923e-dc52-4bb6-a84a-d87c494299ae.png)

**Comment by arthurgousset** [comment]
Hey @aaronmboyd, thanks for posting here. 

Honest answer, unfortunately I don't know the details of the Baklava traffic.
To be transparent, the eng team is currently all hands on deck releasing the new [Federated Attestation product](https://forum.celo.org/t/core-contracts-release-8/4050/5) (i.e. v2 of the attestations protocol). This is a full team effort that will require at least another ~2 weeks of team focus.

Once v2 of the attestations protocol is live, we will shift our attention to v1 again.
Hope that provides so context and sorry I couldn't provide a better answer!

**Reply by aaronmboyd** [reply]
Thanks, appreciate it. Is the code for the attestation bot somewhere I can check into it myself?

**Reply by arthurgousset** [reply]
Yes, but unfortunately it's not very well documented. 

Here is what @eelanagaraj shared of the top of her head:
- [celotool](https://github.com/celo-org/celo-monorepo/tree/master/packages/celotool) is used to deploy the bot
- the Baklava bot runs in a kubernetes instance that cLabs owns and sends a bunch of requests out to attestation services on baklava at some time interval
- It is deployed by celotool, and backtracing from [one of the PRs](https://github.com/celo-org/celo-monorepo/pull/1851/files#diff-48f1433066454777cf9eb7f015a1b6c77706c8131cfcf35a5bb6f21e7c7738f7) that updated that helm chart it looks like this file within celotool is the relevant one: [auto-verify.ts](https://github.com/celo-org/celo-monorepo/blob/cd9a9b5edb4b77ce0854e2b7ec94c803b96f534f/packages/celotool/src/cmds/bots/auto-verify.ts)
